plugins/producer/main.go

```diff
@@ -264,21 +265,31 @@ func InitializeNode(s core.Node, b restricted.Backend) {
 					select {
 					case txEvent := <-ch:
 						for _, txBytes := range txEvent.Txs {
-							// Switch from MarshalBinary to RLP encoding to match EtherCattle's legacy format for txpool transactions
+							// We are using both UnmarshalBinary and RLP encoding to enable both the legacy format and post Cancun formats for txpool transactions
 							tx := &types.Transaction{}
-							if err := tx.UnmarshalBinary(txBytes); err != nil {
-								log.Error("Error unmarshalling")
+							if len(txBytes) == 0 {
 								continue
-							}
-							txdata, err := rlp.EncodeToBytes(tx)
-							if err == nil {
-								select {
-								case producer.Input() <- &sarama.ProducerMessage{Topic: *txPoolTopic, Value: sarama.ByteEncoder(txdata)}:
-								case err := <-producer.Errors():
-									log.Error("Error emitting: %v", "err", err.Error())
-								}
 							} else {
-								log.Warn("Error RLP encoding transactions", "err", err)
+								err := tx.UnmarshalBinary(txBytes)
+								if err == nil {
+									select {
+									case producer.Input() <- &sarama.ProducerMessage{Topic: *txPoolTopicV2, Value: sarama.ByteEncoder(txBytes)}:
```

Comment:
We need to check that `*txPoolTopicV2 != ""`

Sarama, the Kafka producer library we use, doesn't sanity check this, while the Kafka broker assumes that the producer library is responsible for this sanitycheck. None of the native tools for interacting with Kafka will let you delete a topic called `""`, and it becomes a big problem.